frontend-test/unit/workspaceServiceSpec.js

```diff
@@ -1423,6 +1423,30 @@ define([
         var result = workspaceService.checkForMissingValuesInPerformanceTable(performanceTable);
         expect(result).toBeFalsy();
       });
+
+      it('should return true is the effect has a missing value and there is no distribution', function() {
+        var performanceTable = [{
+          performance: {
+            effect: {
+              type: 'empty'
+            }
+          }
+        }];
+        var result = workspaceService.checkForMissingValuesInPerformanceTable(performanceTable);
+        expect(result).toBeTruthy();
+      });
+
+      it('should return true is the distribution has a missing value and there is no effect', function() {
+        var performanceTable = [{
+          performance: {
+            distribution: {
+              type: 'empty'
+            }
+          }
+        }];
+        var result = workspaceService.checkForMissingValuesInPerformanceTable(performanceTable);
+        expect(result).toBeTruthy();
+      });
     });
```

Comment:
missing a test for the false case I think?

Reply:
Second test: 'should return false if each entry has atleast one non-empty effect or distribution'.
Or you think there should be another case?

Reply:
Apologies, looks like code folding messed me up